shared/pingpong/accounts.go

```diff
@@ -667,28 +667,37 @@ func (pps *WorkerState) prepareApps(client *libgoal.Client) (err error) {
 	}
 
 	// generate new apps
+	// cycle through accts and create apps until the desired quantity is reached
 	var txgroup []transactions.Transaction
 	var senders []string
-	for addr, acct := range pps.accounts {
-		if len(pps.cinfo.AppParams) >= int(pps.cfg.NumApp) {
-			break
-		}
-		var tx transactions.Transaction
-		tx, err = pps.newApp(addr, client)
-		if err != nil {
-			return
-		}
-		acct.addBalance(-int64(pps.cfg.MaxFee))
-		txgroup = append(txgroup, tx)
-		senders = append(senders, addr)
-		if len(txgroup) == int(pps.cfg.GroupSize) {
-			pps.schedule(len(txgroup))
-			err = pps.sendAsGroup(txgroup, client, senders)
+	appsPerAddr := make(map[string]int)
+	totalAppCnt := 0
+	for totalAppCnt < int(pps.cfg.NumApp) {
+		for addr, acct := range pps.accounts {
+			if totalAppCnt >= int(pps.cfg.NumApp) {
+				break
+			}
+
+			var tx transactions.Transaction
+			tx, err = pps.newApp(addr, client)
 			if err != nil {
 				return
 			}
-			txgroup = txgroup[:0]
-			senders = senders[:0]
+			acct.addBalance(-int64(pps.cfg.MaxFee))
```

Comment:
```suggestion
			newAppAddrs = append(newAppAddrs, addr)
			acct.addBalance(-int64(pps.cfg.MaxFee))
```
then loop just over those addrs below when fetching account info